mongodbatlas/search.go

```diff
@@ -0,0 +1,233 @@
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+const (
+	searchBasePath = "groups/%s/clusters/%s/fts"
+)
+
+// SearchService provides access to the search related functions in the Atlas API.
+//
+// See more: https://docs.atlas.mongodb.com/reference/api/atlas-search/
+type SearchService interface {
+	ListIndexes(context.Context, string, string, string, string, *ListOptions) ([]*SearchIndex, *Response, error)
+	GetIndex(context.Context, string, string, string) (*SearchIndex, *Response, error)
+	CreateIndex(context.Context, string, string, *SearchIndex) (*SearchIndex, *Response, error)
+	UpdateIndex(context.Context, string, string, string, *SearchIndex) (*SearchIndex, *Response, error)
+	DeleteIndex(context.Context, string, string, string) (*Response, error)
+	ListAnalyzers(context.Context, string, string, *ListOptions) ([]*SearchAnalyzer, *Response, error)
+}
+
+// SearchServiceOp provides an implementation of the SearchService interface
+type SearchServiceOp service
+
+var _ SearchService = &SearchServiceOp{}
+
+// ListIndexes Get all Atlas Search indexes for a specified collection.
+//
+// See more: https://docs.atlas.mongodb.com/reference/api/fts-indexes-get-all/
+func (s *SearchServiceOp) ListIndexes(ctx context.Context, groupID, clusterName, dbName, collName string, opts *ListOptions) ([]*SearchIndex, *Response, error) {
+	if groupID == "" {
+		return nil, nil, NewArgError("GroupID", "must be set")
+	}
+	if clusterName == "" {
+		return nil, nil, NewArgError("ClusterName", "must be set")
+	}
+	if dbName == "" {
+		return nil, nil, NewArgError("DatabaseName", "must be set")
+	}
+	if collName == "" {
+		return nil, nil, NewArgError("CollectionName", "must be set")
+	}
+
+	path := fmt.Sprintf(searchBasePath, groupID, clusterName)
+	path = fmt.Sprintf("%s/indexes/%s/%s", path, dbName, collName)
+
+	path, err := setListOptions(path, opts)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	req, err := s.Client.NewRequest(ctx, http.MethodGet, path, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var root []*SearchIndex
+	resp, err := s.Client.Do(ctx, req, &root)
+
+	return root, resp, err
+}
+
+// Get gets one Atlas Search index by its indexId.
+//
+// See more: https://docs.atlas.mongodb.com/reference/api/fts-indexes-get-one/
+func (s *SearchServiceOp) GetIndex(ctx context.Context, projectID, clusterName, indexID string) (*SearchIndex, *Response, error) {
```

Comment:
[np] Api docs reference group id rather than project id for these endpoints (the above one is group id though)